core/src/test/java/io/questdb/griffin/ImplicitTypeConversionsTest.java

```diff
@@ -0,0 +1,533 @@
+    }
+
+    @Test
+    public void testInsertLongAsByte_Causes_Underflow_and_throws_exception() throws Exception {
+        testInsertCausesException("long", "-129", "byte");
+    }
+    
+    
+    //int->short
+    @Test
+    public void testInsertZeroIntAsShort_ReturnsExactValue() throws Exception {
+        testInsert("int", "0", "short");
+    }
+
+    @Test
+    public void testInsertIntAsShort_ReturnsMinValue() throws Exception {
+        testInsert("int", "-32768", "short");
+    }
+
+    @Test
+    public void testInsertIntAsShort_ReturnsMaxValue() throws Exception {
+        testInsert("int", "32767", "short");
+    }
+
+    @Test
+    public void testInsertIntAsShort_Causes_Overflow_and_throws_exception() throws Exception {
+        testInsertCausesException("int","32768","short");
+    }
+
+    @Test
+    public void testInsertIntAsShort_Causes_Underflow_and_throws_exception() throws Exception {
+        testInsertCausesException("int","-32769","short");
+    }
+
+
+    //int->byte
+    @Test
+    public void testInsertZeroIntAsByte_ReturnsExactValue() throws Exception {
+        testInsert("int","0","byte");
+    }
+
+    @Test
+    public void testInsertIntAsByte_ReturnsMinValue() throws Exception {
+        testInsert("int","-128","byte");
+    }
+
+    @Test
+    public void testInsertIntsByte_ReturnsMaxValue() throws Exception {
+        testInsert("int","127","byte");
+    }
+
+    @Test
+    public void testInsertIntAsByte_Causes_Overflow_and_throws_exception() throws Exception {
+        testInsertCausesException("int", "128", "byte");
+    }
+
+    @Test
+    public void testInsertIntAsByte_Causes_Underflow_and_throws_exception() throws Exception {
+        testInsertCausesException("int", "-129", "byte");
+    }
+
+    
+    //short->byte
+    @Test
+    public void testInsertZeroShortAsByte_ReturnsExactValue() throws Exception {
+        testInsert( "short", "0", "byte" );
+    }
+
+    @Test
+    public void testInsertShortAsByte_ReturnsMinValue() throws Exception {
+        testInsert( "short", "-128", "byte" );
+    }
+
+    @Test
+    public void testInsertShortAsByte_ReturnsMaxValue() throws Exception {
+        testInsert( "short", "127", "byte" );
+    }
+
+    @Test
+    public void testInsertShortAsByte_Causes_Overflow_and_throws_exception() throws Exception {
+        testInsertCausesException( "short", "128", "byte" );
+    }
+
+    @Test
+    public void testInsertShortAsByte_Causes_Underflow_and_throws_exception() throws Exception {
+        testInsertCausesException( "short", "-129", "byte" );
+    }
+    
+    private void testInsert(String valueType, String value, String targetColumnType, String expectedValue) throws Exception {
+        assertMemoryLeak(() -> {
+            compiler.compile("create table tab(x " + targetColumnType + " );", sqlExecutionContext);
+            executeInsert("insert into tab values (cast(" + value + " as " + valueType + " ));");

```

Comment:
if `cast` is what we are testing, then we do not need to create/insert into a table.

Example from `CastNullFunctionFactoryTest`:

```
@Test
    public void testCastNullToNonExisingTypeShouldFail() {
        try {
            assertQuery(null, "cast(null as NON_EXISTING_TYPE)", null, null);
            Assert.fail();
        } catch (Exception expected) {
            Assert.assertEquals("[13] invalid constant: NON_EXISTING_TYPE", expected.getMessage());
        }
    }
```

Reply:
Thanks for the suggestion but it seems range check shouldn't trigger for explicit casts .

Reply:
yes, ignore this please, we need the asm bit to be hit.